packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-has-one-relationship/BookUpdateForm.jsx

```diff
@@ -0,0 +1,391 @@
+            Book.copyOf(original, (updated) => {
+              Object.assign(updated, modelFields);
+            }),
+          );
+          if (onSuccess) {
+            onSuccess(modelFields);
+          }
+        } catch (err) {
+          if (onError) {
+            onError(modelFields, err.message);
+          }
+        }
+      }}
+      {...rest}
+      {...getOverrideProps(overrides, 'BookUpdateForm')}
+    >
+      <TextField
+        label="Name"
+        isRequired={false}
+        isReadOnly={false}
+        onChange={(e) => {
+          let { value } = e.target;
+          if (onChange) {
+            const modelFields = {
+              name: value,
+              primaryAuthor,
+            };
+            const result = onChange(modelFields);
+            value = result?.name ?? value;
+          }
+          if (errors.name?.hasError) {
+            runValidationTasks('name', value);
+          }
+          setName(value);
+        }}
+        onBlur={() => runValidationTasks('name', name)}
+        errorMessage={errors.name?.errorMessage}
+        hasError={errors.name?.hasError}
+        {...getOverrideProps(overrides, 'name')}
+      ></TextField>
+      <ArrayField
+        lengthLimit={1} // new prop
+        getBadgeText={getDisplayValue['primaryAuthor']} // new prop
+        onChange={async (items) => {
+          let value = items[0]; // function changed for single value
+          if (onChange) {
+            const modelFields = {
+              name,
+              primaryAuthor: value,
+            };
+            const result = onChange(modelFields);
+            value = result?.primaryAuthor ?? value;
+          }
+          setPrimaryAuthor(value);
+          setCurrentPrimaryAuthorValue(undefined); // both value and displayValue emptied
+          setCurrentPrimaryAuthorDisplayValue(undefined);
+        }}
+        currentFieldValue={currentPrimaryAuthorValue}
+        label={'Primary author'}
+        items={[primaryAuthor]} // passed in as array
+        hasError={errors.primaryAuthor?.hasError}
+        setFieldValue={setCurrentPrimaryAuthorDisplayValue} // display value must be set on edit
+        inputFieldRef={primaryAuthorRef}
+        defaultFieldValue={undefined}
+      >
+        <Autocomplete
+          label="Primary author"
+          isRequired={false}
+          isReadOnly={false}
+          onChange={(e) => {
+            // called when user is typing
+            let { value } = e.target;
+            if (errors.primaryAuthor?.hasError) {
+              runValidationTasks('primaryAuthor', value);
+            }
+            setCurrentPrimaryAuthorDisplayValue(value);
+            setCurrentPrimaryAuthorValue(undefined); // empty out prev selected option
+          }}
+          suggestions={authorRecords.map((r) => ({
+            id: r.id,
+            label: getDisplayValue['primaryAuthor']?.(record) ?? r.id,
+          }))}
+          onSuggestionSelect={({ id, label }) => {
```

Comment:
And also this will simply be `onSelect` now

Reply:
Updated, thanks